### DISCUSSION TITLE
Databind  2.15.2 and  DeserializationFeature.FAIL_ON_TRAILING_TOKENS

### DISCUSSION THREAD

**Discussion by mebigfatguy** [FasterXML/jackson-databind#4143] [Q&A]
Hi folks, 

i'm trying to upgrade jackson-databind 2.14.2 ->2.15.2 on a java webapp, and after updating,  i'm getting scores of 

```
Trailing token (of type FIELD_NAME) found after value (bound as `com.acme.pgbu.rest.resources.sec.UserAdministrationRO`)
```

I'm assuming  that  DeserializationFeature.FAIL_ON_TRAILING_TOKENS is now turned on by default in 2.15.2, and perhaps i could  just disable that, but trying to figure out how to debug what  is going on  and why this is happening. On the  client  side the request body is
```
{"projectDirectoryUserGroups":{"added":[{"description":"padQCtXgzffq7csjU0PgMtT4pyKFkOmeaTXBRGHFEHUf1j7EeTKQme0lO4972tsiF3pcKDgWDbYbRZcbosCZtuz3pohVuDfenTMM0uncv6djOLUELginiCl5jSgFJW4PhEpef52O8fv4rbrKupo3HhQOCl8dMh06r5JY2qw2DWGoNUYORznItzesyPjTWUfQrIERR97Ga37EDLDAqc2bEWRXB8pgfmno0b7CU0kLEqYpkqVBuk7GQJdtwcEggNt","assignGroup":true,"id":-1,"displayName":"ProjTeamUserBHCjm3APoKoP3TGWscvrocSuF6aACnSlk1Dt40fu6h5Gh1dtN7","defaultprofiles":{"PROJECT":{"description":"EiGld6ck8GAUeiX2DQADYy5OqJ","internalName":"26","privileges":[{"resourceType":"PROJECTS","actions":["edit"],"resourceURI":"TEMPLATE"},{"resourceType":"PROJECTS","actions":["delete"],"resourceURI":"TEMPLATE"}],"displayName":"IntgProjectDefaultGroupc2d3Fs","isAdmin":false,"readonly":false,"profileId":3,"name":"IntgProjectDefaultGroupc2d3Fs","default":false}},"roleMembers":[{"type":"User","guid":"83b1b215cc8141b4892285d4b349e424","realUserName":"Intg-acme-phx-1187VdaFsu@oracle.foo","displayName":"Intg-acme-phx-1187VdaFsu@oracle.foo","name":"Intg-acme-phx-1187VdaFsu@oracle.foo"},{"type":"User","guid":"ea75b57dcd474eba9b2e37b30e6e449f","realUserName":"Intg-acme-phx-1187XXFoQS@oracle.foo","displayName":"Intg-acme-phx-1187XXFoQS@oracle.foo","name":"Intg-acme-phx-1187XXFoQS@oracle.foo"}],"groupId":-1,"inherited":false,"seeded":false}],"deleted":[],"modified":[]},"projectDirectoryUserGroupUsers":null}
```

(the data is junk because  it's actually an integration test). But  It 'looks' ok on the client side,and i don't get a  chance to  see it on the server side because the exception happens.

I was wondering if there's  some way to turn on extra logging or debugging tips to see what the body looks like  on the server side, or  even more useful, what is the offending  verbiage?

**Comment by JooHyukKim** [comment]
> I'm assuming that DeserializationFeature.FAIL_ON_TRAILING_TOKENS is now turned on by default in 2.15.2

No, the feature is still disabled by default in 2.15. Please check the [🔗 official documentation](https://javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/latest/index.html), no need to rely on assumptions 👍🏼. Plus such backward-**incompatible** change would not be attempted by the community with enough consideration and articulate transitions.

> I was wondering if there's some way to turn on extra logging or debugging tips to see what the body looks like on the server side, or even more useful, what is the offending verbiage?

Have you tried catching the exception thrown by Jackson? What type of exception is thrown? I think there is enough information provided by the exception. For example, `MismatchedInputException#getLocation`

Hope it helps.
Cheers.

**Comment by cowtowncoder** [comment]
As per @JooHyukKim no change in Jackson wrt defaults. I'd suspect framework (Spring, Quarkus) possibly changing defaults in their settings, if you have upgraded those dependencies.

**Comment by mebigfatguy** [comment]
This  is literally all i changed (the fasterxml version)

The http request returns a HttpStatus.SC_BAD_REQUEST with the body being

```
Trailing token (of type FIELD_NAME) found after value (bound as `com.acme.pgbu.rest.resources.sec.UserAdministrationRO`):
not allowed as per `DeserializationFeature.FAIL_ON_TRAILING_TOKENS`  at [Source: 
(org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream); line: 1, column: 1323]
```

As  for catching an exception, how would one  do that? i assume  the exception happens before any code in  the endpoint is called??

```
@Path("/{projectId}/userGroups")
    @Consumes(MediaType.APPLICATION_JSON)
    public UserAdministrationRO processGroup(UserAdministrationRO userAdminService, @PathParam("projectId") Long projectId) {
``` 
  
can't set a break point here,  never reaches, as the exception happens as userAdminService attempts  to  be populated from body?

**Reply by cowtowncoder** [reply]
Hmmh. Reproduction would be good but since it's via framework, probably needs to be reported to that project.

As to catching, that's up to framework as well. You can typically take in, say, `byte[]` and use `ObjectMapper.readValue()` directly if you want. But otherwise it depends on ExceptionMapper (etc) that framework provides, not much you can do via Jackson itself. And yes, exception by framework thrown before endpoint code called.
